src/cdk-experimental/listbox/listbox.ts

```diff
@@ -7,98 +7,214 @@
 
 let nextId = 0;
 
-/**
- * Directive that applies interaction patterns to an element following the aria role of option.
- * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
- * value is built in.
- */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
-    role: 'option',
+    'role': 'option',
     '(click)': 'toggle()',
-    '[attr.aria-selected]': 'selected || null',
+    '(focus)': 'activate()',
+    '(blur)': 'deactivate()',
     '[id]': 'id',
+    '[attr.aria-selected]': '_selected || null',
+    '[attr.tabindex]': '_getTabIndex()',
+    '[attr.aria-disabled]': '_isInteractionDisabled()',
+    '[class.cdk-option-disabled]': '_isInteractionDisabled()',
+    '[class.cdk-option-active]': '_active'
+
   }
 })
-export class CdkOption {
+export class CdkOption implements ListKeyManagerOption, Highlightable {
   private _selected: boolean = false;
+  private _disabled: boolean = false;
+  _active: boolean = false;
 
-  /** Whether the option is selected or not */
   @Input()
   get selected(): boolean {
     return this._selected;
   }
   set selected(value: boolean) {
-    this._selected = coerceBooleanProperty(value);
+    if (!this._disabled) {
+      this._selected = coerceBooleanProperty(value);
+    }
   }
 
-  /** The id of the option, set to a uniqueid if the user does not provide one */
+  /** The id of the option, set to a uniqueid if the user does not provide one. */
   @Input() id = `cdk-option-${nextId++}`;
 
-  constructor(@Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {}
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  constructor(private _elementRef: ElementRef,
+              @Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {
+  }
 
-  /** Toggles the selected state, emits a change event through the injected listbox */
+  /** Toggles the selected state, emits a change event through the injected listbox. */
   toggle() {
-    this.selected = !this.selected;
-    this.listbox._emitChangeEvent(this);
+    if (!this._isInteractionDisabled()) {
+      this.selected = !this.selected;
+      this.listbox._emitChangeEvent(this);
+    }
+  }
+
+  /** Sets the active property true if the option and listbox aren't disabled. */
+  activate() {
+    if (!this._isInteractionDisabled()) {
+      this._active = true;
+      this.listbox.setActiveOption(this);
+    }
+  }
+
+  /** Sets the active property false. */
+  deactivate() {
+    this._active = false;
+  }
+
+  /** Returns true if the option or listbox are disabled, and false otherwise. */
+  _isInteractionDisabled(): boolean {
+    return (this.listbox.disabled || this._disabled);
+  }
+
+  /** Returns the tab index which depends on the disabled property. */
+  _getTabIndex(): string | null {
+    return (this.listbox.disabled || this._disabled) ? null : '-1';
```

Comment:
I'm not quite understanding why it's necessary to remove the -1 tabindex when the control is disabled, since -1 also prevents user tabbing. Is the intent to remove focus from an option when it becomes disabled?

Reply:
Yes so the way it will work is that the option is skipped in the tab order. I believe this was in one of the specs but I will check.